port/unix/omrsysinfo.c

```diff
@@ -1967,7 +1974,194 @@ omrsysinfo_get_linux_aarch64_description(struct OMRPortLibrary *portLibrary, OMR
 
 	return 0;
 }
-#endif /* defined(AARCH64) && defined(LINUX) */
+#elif defined(OSX) /* defined(LINUX) */
+/**
+ * @internal
+ * Populates OMRProcessorDesc *desc on macOS ARM64.
+ *
+ * @param[in] portLibrary The port library.
+ * @param[in] desc pointer to the struct that will contain the CPU type and features.
+ *
+ * @return 0 on success.
+ */
+static intptr_t
+omrsysinfo_get_macos_aarch64_description(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc)
+{
+	int64_t val = 0;
+	size_t size = sizeof(val);
+
+	desc->processor = OMR_PROCESSOR_ARM64_V8_A;
+	desc->physicalProcessor = desc->processor;
+	desc->features[0] = 0;
+	desc->features[1] = 0;
+
+	/*
+	 * These features are standard on Apple Silicon
+	 * No need to check
+	 */
+	omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_FP);
+	omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_ASIMD);
+
+	/*
+	 * Advanced SIMD and Floating Point Capabilities
+	 */
+	if ((0 == sysctlbyname("hw.optional.arm.FEAT_BF16", &val, &size, NULL, 0))
+	    && (0 != val)
+	) {
+		omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_BF16);
+	}
+
+	if ((0 == sysctlbyname("hw.optional.arm.FEAT_DotProd", &val, &size, NULL, 0))
+	    && (0 != val)
+	) {
+		omrsysinfo_set_feature(desc, OMR_FEATURE_ARM64_DOTPROD);
+	}
+
+	/* Attribute name changed in macOS 12. */
+	if (((0 == sysctlbyname("hw.optional.arm.FEAT_FCMA", &val, &size, NULL, 0))
+	     || (0 == sysctlbyname("hw.optional.armv8_3_compnum", &val, &size, NULL, 0)))
+	    && (0 != val)
```

Comment:
There is an alignment issue since tabs and spaces are mixed. This is observed in all `if` statements within this function. We should not mix tabs and spaces. We should just use tabs. See https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#indent-code-using-tabs-not-spaces.
```suggestion
	if (((0 == sysctlbyname("hw.optional.arm.FEAT_FCMA", &val, &size, NULL, 0))
	|| (0 == sysctlbyname("hw.optional.armv8_3_compnum", &val, &size, NULL, 0)))
	&& (0 != val)
```

Reply:
Fixed.